proposal/jptom/README.rst

```diff
@@ -0,0 +1,99 @@
+===========================
+Proposal for Kalman filter 
+===========================
+
+Basic information
+=================
+
+`My project repository <https://github.com/jptom/study-probablistic-robotics/tree/main/nsd-term-project>`_
+
+My base python code is from this book.
+This author translated `Probabilistic Robotics <https://www.amazon.com/Probabilistic-Robotics-INTELLIGENT-ROBOTICS-AUTONOMOUS-ebook/dp/B00DJD9LXC>`_ in 2005.
+This book introduce it's implementation in python by him in 2019.
+
+.. image:: https://images-na.ssl-images-amazon.com/images/I/91C20m7ZgoL.jpg
+   :scale: 20%
+   
+Problem to solve
+================
+
+
+.. image:: image/kalmansim.PNG
+.. image:: image/kalman.PNG
+
+Kalman filter is a technique of self-localazation of robotics.
+It estimale mean and covarience of Gaussian distribution using observation of location of landmarks.
+I am planning to implement whole the calculation as one function in c++.
+   
+Perspective users
+=================
+
+Beginner of probabilistic robotics
+
+System architecture
+===================
+
+This is the simulator architecture.
+.. image:: image/architecture.jpg
+
+Kalman filter is a part of "one step" function of robot object.
+The World call "one step" function then all "one step" function of object who is 
+registered in The World will be called.
+
+API description
+===============
+
+Users need to write own robot program which has "one step" and "draw" function.
+My package will support two popular self-localazation algorithm, one is Kalman filter.
+I want the package can be use in not only simulator but also real robot.
+
+.. code-block:: python
+
+   class Robot:
+      def one_step{ 
+         user code 
+         filter.observation_update()
+      }
+     
+      def draw{ user code }
+     
+   class Filter:
+      def observation_update(observation){
+         Implement in C++
+      }
+
+Engineering infrastructure
+==========================
+
+The package build with cmake. 
+
+The testing use simulation.
```

Comment:
How do you plan for automatic testing?  You mentioned that a simulation may take non-trivial time (seconds).  Would it be possible to decompose the simulation so that you can test each of the parts separately?